source_control/gitlab_tag.py

```diff
@@ -0,0 +1,268 @@
+            validate_certs=dict(required=False, default=True, type='bool', aliases=['verify_ssl']),
+            login_user=dict(required=False, no_log=True),
+            login_password=dict(required=False, no_log=True),
+            login_token=dict(required=False, no_log=True),
+            group=dict(required=True),
+            project=dict(required=True),
+            name=dict(required=True),
+            ref=dict(required=False, default="master"),
+            description=dict(required=False),
+            state=dict(default="present", choices=["present", 'absent']),
+        ),
+        supports_check_mode=True
+    )
+
+    if not HAS_GITLAB_PACKAGE:
+        module.fail_json(msg="Missing required gitlab module (check docs or install with: pip install python-gitlab")
+
+    server_url = module.params['server_url']
+    verify_ssl = module.params['validate_certs']
+    login_user = module.params['login_user']
+    login_password = module.params['login_password']
+    login_token = module.params['login_token']
+    project = module.params['project']
+    group = module.params['group']
+    name = module.params['name']
+    ref = module.params['ref']
+    description = module.params['description']
+    state = module.params['state']
+    use_credentials = None
+
+    # Validate some credentials configuration parameters.
+    if login_user is not None and login_password is not None:
+        use_credentials = True
+    elif login_token is not None:
+        use_credentials = False
+    else:
+        module.fail_json(msg="No login credentials are given. Use login_user with login_password, or login_token")
+
+    if login_token and login_user:
+        module.fail_json(msg="You can either use 'login_token' or 'login_user' and 'login_password'")
+
+    try:
+        if use_credentials:
+            git = gitlab.Gitlab(server_url, email=login_user, password=login_password, ssl_verify=verify_ssl)
+            git.auth()
+        else:
+            git = gitlab.Gitlab(server_url, private_token=login_token, ssl_verify=verify_ssl)
+            git.auth()
+    except Exception:
+        e = get_exception()
+        module.fail_json(msg="Failed to connect to Gitlab server: %s " % e)
+
+    tag = GitLabTag(module, git)
+    project = tag.getProject(group=group, project=project)
+    if not project:
+        module.fail_json(msg="The project or the group does not exists. We can not create a tag.")
+
+    project_tag = tag.existsTag(name=name)
+    if project_tag:
+        if state == "present":
+            module.exit_json(changed=False, result="Tag is already created.")
+        else:
+            if module.check_mode:
+                module.exit_json(changed=True, result="Tag should have been deleted.")
+            if tag.deleteTag():
+                module.exit_json(changed=True, result="Tag is deleted.")
+            else:
+                module.exit_json(changed=False, result="Something went wrong deleting the tag.")
```

Comment:
Is there a way to get a error message or something, as a user would be puzzled by this error message (maybe we can't, I didn't looked at the code)